PhysicsTools/NanoAOD/plugins/IsoValueMapProducer.cc

```diff
@@ -0,0 +1,318 @@
+// -*- C++ -*-
+//
+// Package:    PhysicsTools/NanoAOD
+// Class:      IsoValueMapProducer
+// 
+/**\class IsoValueMapProducer IsoValueMapProducer.cc PhysicsTools/NanoAOD/plugins/IsoValueMapProducer.cc
+
+ Description: [one line class summary]
+
+ Implementation:
+     [Notes on implementation]
+*/
+//
+// Original Author:  Marco Peruzzi
+//         Created:  Mon, 04 Sep 2017 22:43:53 GMT
+//
+//
+
+
+// system include files
+#include <memory>
+
+// user include files
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/global/EDProducer.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Utilities/interface/StreamID.h"
+
+#include "RecoEgamma/EgammaTools/interface/EffectiveAreas.h"
+
+#include "DataFormats/PatCandidates/interface/Muon.h"
+#include "DataFormats/PatCandidates/interface/Electron.h"
+#include "DataFormats/PatCandidates/interface/Photon.h"
+#include "DataFormats/PatCandidates/interface/IsolatedTrack.h"
+
+//
+// class declaration
+//
+
+template <typename T>
+class IsoValueMapProducer : public edm::global::EDProducer<> {
+   public:
+  explicit IsoValueMapProducer(const edm::ParameterSet &iConfig):
+    src_(consumes<edm::View<T>>(iConfig.getParameter<edm::InputTag>("src")))
+  {
+    if ((typeid(T) == typeid(pat::Muon)) || (typeid(T) == typeid(pat::Electron)) || typeid(T) == typeid(pat::IsolatedTrack)) {
+      produces<edm::ValueMap<float>>("miniIsoChg");
+      produces<edm::ValueMap<float>>("miniIsoAll");
```

Comment:
not Nano-specific.
Move to PhysicsTools/PatAlgos? 
[perhaps PatAlgos is getting too large , maybe split off these "skimmers" to PhysicsTools/ReducedPatAlgos]